### PR TITLE
List sales record implementation

### PR DESCRIPTION
**Pull request by nickmcm92**
Retrieves sales records, including total price and displays in a table.
To view the sales records, please visit [http://localhost:8080/](http://localhost:8080/)

### CODE REVIEW

**Comment by nickmcm92** [comment]
Converting to draft to implement sorting and search

**Comment by nickmcm92** [comment]
Marking PR as ready for review. Will implement search/date filtering in separate PR

File: src/js/list.js [hunk 1]
```diff
@@ -1,9 +1,115 @@
...
+        let row = document.createElement("tr");
+        let recordLink = document.createElement("a");
+        recordLink.href = "/view.php?id=" + record.SalesRecordNumber;
+        let recordNum = document.createElement("td");
+        let recordDate = document.createElement("td");
+        let recordTotal = document.createElement("td");
+
+        recordLink.innerHTML = record.SalesRecordNumber;
+        recordNum.appendChild(recordLink.cloneNode(true));
+        recordLink.innerHTML = record.SalesDate; //TODO: Adjust to a better date
```
Comment (hd-chuong) [inline src/js/list.js]: An option to adjust this sales date could be using momentJS to format the date.

 `recordLink.innerHTML = moment(record.SalesDate).format("YYYY-MM-DD");`

File: src/js/list.js [hunk 2]
```diff
@@ -1,9 +1,115 @@
...
+        pageItem.appendChild(link);
+        paginationHolder.appendChild(pageItem);
+    }
+
+    //create each record
+    let recordTable = $('tbody')[0];
+    for(const record of body.results) {
+        let row = document.createElement("tr");
+        let recordLink = document.createElement("a");
+        recordLink.href = "/view.php?id=" + record.SalesRecordNumber;
```
Comment (hd-chuong) [inline src/js/list.js]: I used recordID. We might need to agree on the  choice of keyword.

**Review by hd-chuong — APPROVED** [review]

File: src/js/list.js (hunk 2 above)
Comment (nickmcm92) [reply src/js/list.js]: I have updated as advised

File: src/js/list.js (hunk 1 above)
Comment (nickmcm92) [reply src/js/list.js]: I have added as you recommended.

**Pull request merged**